torch/_dynamo/variables/builtin.py

```diff
@@ -1173,6 +1174,50 @@ def call_setattr(
             tx.output.side_effects.is_attribute_mutation(obj)
             and name_var.is_python_constant()
         ):
+            if isinstance(obj, variables.TensorVariable):
+                from .builder import wrap_fx_proxy
+
+                if name_var.as_python_constant() == "data":
+                    # Remove the old reference in tracked fakes - if we don't do this
+                    # new .data value size and shape differences will cause
+                    # tracked fakes to produce incorrect guards. This is sound because the TensorVariable
+                    # coming out of set_() below will be a new one, and get
+                    # installed in tracked fakes.
+                    to_remove = []
+                    for tf in tx.output.tracked_fakes:
+                        if tf.source == obj.source:
+                            to_remove.append(tf)
+                    for tf in to_remove:
+                        tx.output.tracked_fakes.remove(tf)
+
+                    # Step 1 - disable grad
+                    version = obj.as_proxy().node.meta["example_value"]._version
+                    with dynamo_disable_grad(tx), torch.no_grad():
+                        # Step 2 - call `set_`
+                        out = wrap_fx_proxy(
+                            tx,
+                            tx.output.create_proxy(
+                                "call_function",
+                                torch.Tensor.set_,
+                                *proxy_args_kwargs([obj, val], {}),
+                            ),
+                        )
+
+                    # Step 3 - drop the version counter - this is a hack required to get
+                    # .data setting to play correctly with the autograd engine.
+                    if version > 0:
+                        verion = version - 1
+                    tx.output.create_proxy(
+                        "call_function",
+                        torch._C._autograd._unsafe_set_version_counter,
+                        (out.as_proxy(), version),
+                        {},
+                    )
```

Comment:
Dynamo isn't obligated to produce aten ops in its FX graph. Why do we not generate a call to a function like `set_data` which internally is implemented as `x.data = y`? I feel you had tried this before and it didn't work but I can't tell from the PR why it doesn't.

Reply:
I found old notes I wrote ~June, it says It makes the autograd engine choke during decomp? but let me give it a whirl, I don't remember and this reads kinda sus.

Reply:
So what happened when you tried it? You didn't respond to the comment. This is very important, because if this alternate strategy works, we can remove *a lot* of code from this PR.

Reply:
Oh, this didn't work in aot, but before I had a bunch of changes. I can check again. However, it doesn't remove that much code? Just some dynamo internal handling. The fakification policy stuff stays exactly the same?

Reply:
It passes silently now, and produces the wrong numbers. This is probably because we do a bunch to make set_ work and track as a mutating op, but this function has none of those traits.